numpy/core/tests/test_function_base.py

```diff
@@ -29,6 +29,30 @@ def test_dtype(self):
         assert_equal(y.dtype, dtype('int32'))
 
 
+class TestGeomspace(TestCase):
+
+    def test_basic(self):
+        y = geomspace(1, 1e6)
+        assert_(len(y) == 50)
+        y = geomspace(1, 1e6, num=100)
+        assert_(y[-1] == 10 ** 6)
+        y = geomspace(1, 1e6, endpoint=False)
+        assert_(y[-1] < 10 ** 6)
+        y = geomspace(1, 1e6, num=7)
+        assert_array_equal(y, [1, 10, 100, 1e3, 1e4, 1e5, 1e6])
+
+    def test_dtype(self):
+        y = geomspace(1, 1e6, dtype='float32')
+        assert_equal(y.dtype, dtype('float32'))
+        y = geomspace(1, 1e6, dtype='float64')
+        assert_equal(y.dtype, dtype('float64'))
+        y = geomspace(1, 1e6, dtype='int32')
+        assert_equal(y.dtype, dtype('int32'))
+
```

Comment:
You may also want to add basically the same tests as `linspace`/`logspace` do, possibly including the ones for subclasses, etc. While currently the tests of `logspace` cover most of the functionality of your function, there is no guarantee that it will be implemented in terms of `logspace` in the future. For example, the `exp(linspace(...))` implementation given by @rkern [here](https://github.com/numpy/numpy/issues/7255#issuecomment-184586446). `numpy.percentile` used to be implemented in terms of `sort`. Then it got reimplemented in terms of `numpy.partition`. That is the sort of thing tests are really good for.